src/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs

```diff
@@ -219,45 +219,72 @@ private SafeProcessHandle GetProcessHandle()
             return new SafeProcessHandle(_processId);
         }
 
-        /// <summary>Starts the process using the supplied start info.</summary>
+        /// <summary>
+        /// Starts the process using the supplied start info. 
+        /// Even with UseShellExecute option, we try first running fileName just in case the caller is giving executable which we should run
+        /// Then if we couldn't run it we'll try the shell tools to launch it(e.g. "open fileName")
+        /// </summary>
         /// <param name="startInfo">The start info with which to start the process.</param>
         private bool StartCore(ProcessStartInfo startInfo)
         {
             string filename;
             string[] argv;
 
+            if (Directory.Exists(startInfo.FileName))
```

Comment:
> if (Directory.Exists(startInfo.FileName)) [](start = 11, length = 42)

please open issue to track if we can open the directory on Linux when having ShellExecute is set. thanks.

Reply:
OK. Just a quick summary here for the reviewers:

- In Linux, xdg-open cannot open folder, so I decided to throw when startInfo.FileName is a directory, regardless of UseShellExecute option being true or false. Ideally we were looking at throwing only when UseShellExecute option was false.

- In OSX, the program "open" successfully opens folder using "finder". But Linux needs extra effort to be able to open a folder here.

Reply:
ok. just log an issue to track looking at that on Linux later.

Reply:
This will throw for OSX and Linux right? 

Should we throw only in Linux as in OSX it works fine, or are we trying to be consistent?

Reply:
@tarekgh issue #23969 now tracks this. 
@safern Yes, I am throwing this way to make up for a consistent behavior. Otherwise in OSX it should fine. (Open uses Finder to open a folder.)